server/apis/v1/handler.go

```diff
@@ -46,282 +68,831 @@ type handler struct {
 	if err != nil {
 		return nil, fmt.Errorf("failed to get kubeclient, %w", err)
 	}
-	metricsClient := metricsversiond.NewForConfigOrDie(restConfig)
-	numaflowClient := dfv1versiond.NewForConfigOrDie(restConfig).NumaflowV1alpha1()
+	metricsClient := metricsversiond.NewForConfigOrDie(k8sRestConfig)
+	numaflowClient := dfv1versiond.NewForConfigOrDie(k8sRestConfig).NumaflowV1alpha1()
 	return &handler{
 		kubeClient:     kubeClient,
 		metricsClient:  metricsClient,
 		numaflowClient: numaflowClient,
 	}, nil
 }
 
+// ListNamespaces is used to provide all the namespaces that have numaflow pipelines running
+func (h *handler) ListNamespaces(c *gin.Context) {
+	namespaces, err := getAllNamespaces(h)
+	if err != nil {
+		h.respondWithError(c, fmt.Sprintf("Failed to fetch all namespaces, %s", err.Error()))
+		return
+	}
+	c.JSON(http.StatusOK, NewNumaflowAPIResponse(nil, namespaces))
+}
+
+// GetClusterSummary summarizes information of all the namespaces in a cluster and wrapped the result in a list.
+func (h *handler) GetClusterSummary(c *gin.Context) {
+	namespaces, err := getAllNamespaces(h)
+	if err != nil {
+		h.respondWithError(c, fmt.Sprintf("Failed to fetch cluster summary, %s", err.Error()))
+		return
+	}
+	var clusterSummary ClusterSummaryResponse
+	// Loop over the namespaces to get status
+	for _, ns := range namespaces {
+		// Fetch pipeline summary
+		pipelines, err := getPipelines(h, ns)
+		if err != nil {
+			h.respondWithError(c, fmt.Sprintf("Failed to fetch cluster summary, %s", err.Error()))
+			return
+		}
+
+		var pipeSummary PipelineSummary
+		var pipeActiveSummary ActiveStatus
+		// Loop over the pipelines and get the status
+		for _, pl := range pipelines {
+			if pl.Status == PipelineStatusInactive {
+				pipeSummary.Inactive++
+			} else {
+				pipeActiveSummary.increment(pl.Status)
+
+			}
+		}
+		pipeSummary.Active = pipeActiveSummary
+
+		// Fetch ISB service summary
+		isbSvcs, err := getIsbServices(h, ns)
+		if err != nil {
+			h.respondWithError(c, fmt.Sprintf("Failed to fetch cluster summary, %s", err.Error()))
+			return
+		}
+
+		var isbSummary IsbServiceSummary
+		var isbActiveSummary ActiveStatus
+		// loop over the ISB services and get the status
+		for _, isb := range isbSvcs {
+			if isb.Status == ISBServiceStatusInactive {
+				isbSummary.Inactive++
+			} else {
+				isbActiveSummary.increment(isb.Status)
+			}
+		}
+		isbSummary.Active = isbActiveSummary
+		clusterSummary = append(clusterSummary, NewClusterSummary(ns, pipeSummary, isbSummary))
+	}
+	c.JSON(http.StatusOK, NewNumaflowAPIResponse(nil, clusterSummary))
+
+}
+
+// CreatePipeline is used to create a given pipeline
+func (h *handler) CreatePipeline(c *gin.Context) {
+	ns := c.Param("namespace")
+	// dryRun is used to check if the operation is just a validation or an actual create
+	dryRun := c.DefaultQuery("dry-run", "false")
+	dryRun = strings.ToLower(dryRun)
+
+	reqBody, err := parseSpecFromReq(c, SpecTypePipeline)
```

Comment:
Since the `Bind()` function has almost same lines of code, I rather see it to be used directly here, than do a type assertion.